rectools/model_selection/random_split.py

```diff
@@ -0,0 +1,199 @@
+    ) -> tp.Iterator[tp.Tuple[np.ndarray, np.ndarray, tp.Dict[str, tp.Any]]]:
+        """
+        Split interactions into folds.
+
+        Parameters
+        ----------
+        interactions: Interactions
+            User-item interactions.
+        collect_fold_stats: bool, default False
+            Add some stats to fold info,
+            like size of train and test part, number of users and items.
+
+        Returns
+        -------
+        iterator(array, array, dict)
+            Yields tuples with train part row numbers, test part row numbers and fold info.
+        """
+        df = interactions.df
+        idx = pd.RangeIndex(0, len(df))
+        test_part_size = len(df) - int(self.train_size * len(df))
+
+        need_ui = self.filter_cold_users or self.filter_cold_items or self.filter_already_seen or collect_fold_stats
+        fold_info = {}
+
+        test_mask = np.zeros_like(idx, dtype=bool)
+        choose_idx = np.random.choice(idx, test_part_size, replace=False)
+        test_mask[choose_idx] = True
+        train_mask = ~test_mask
+
+        train_idx = idx[train_mask].values
+        test_idx = idx[test_mask].values
+
+        if need_ui:
+            train_users = df[Columns.User].values[train_mask]
+            train_items = df[Columns.Item].values[train_mask]
+            test_users = df[Columns.User].values[test_mask]
+            test_items = df[Columns.Item].values[test_mask]
+
+        unq_train_users = None
+        unq_train_items = None
+
+        if self.filter_cold_users:
+            unq_train_users = pd.unique(train_users)
+            mask = np.isin(test_users, unq_train_users)
+            test_users = test_users[mask]
+            test_items = test_items[mask]
+            test_idx = test_idx[mask]
+
+        if self.filter_cold_items:
+            unq_train_items = pd.unique(train_items)
+            mask = np.isin(test_items, unq_train_items)
+            test_users = test_users[mask]
+            test_items = test_items[mask]
+            test_idx = test_idx[mask]
+
+        if self.filter_already_seen:
+            mask = get_not_seen_mask(train_users, train_items, test_users, test_items)
+            test_users = test_users[mask]
+            test_items = test_items[mask]
+            test_idx = test_idx[mask]
+
+        if collect_fold_stats:
+            if unq_train_users is None:
+                unq_train_users = pd.unique(train_users)
+            if unq_train_items is None:
+                unq_train_items = pd.unique(train_items)
+
+            fold_info["Train"] = train_users.size
+            fold_info["Train users"] = unq_train_users.size
+            fold_info["Train items"] = unq_train_items.size
+            fold_info["Test"] = test_users.size
+            fold_info["Test users"] = pd.unique(test_users).size
+            fold_info["Test items"] = pd.unique(test_items).size
+
+        yield train_idx, test_idx, fold_info
+
+
+def get_not_seen_mask(
```

Comment:
Of course, you don't need to copy this function from other module.

It's better to extract it to a separate module, say `rectools/model_selection/utils.py`, and import in both modules.